src/ring_theory/finiteness.lean

```diff
@@ -126,6 +126,22 @@ lemma _root_.subalgebra.fg_bot_to_submodule {R A : Type*}
   (⊥ : subalgebra R A).to_submodule.fg :=
 ⟨{1}, by simp [algebra.to_submodule_bot] ⟩
 
+lemma fg_unit {R A : Type*} [comm_semiring R] [semiring A] [algebra R A]
+  (I : (submodule R A)ˣ) : (I : submodule R A).fg :=
+begin
+  have : (1 : A) ∈ (I * ↑I⁻¹ : submodule R A),
+  { rw units.mul_inv, exact one_le.mp le_rfl },
+  obtain ⟨T, T', hT, hT', one_mem⟩ := mem_span_mul_finite_of_mem_mul this,
+  refine ⟨T, span_eq_of_le _ hT _⟩,
+  rw [← one_mul ↑I, ← mul_one (span R ↑T)],
+  conv_rhs { rw [← units.inv_mul I, ← mul_assoc] },
+  refine mul_le_mul_left (le_trans _ $ mul_le_mul_right $ span_le.mpr hT'),
+  rwa [one_le, span_mul_span],
+end
+
+lemma fg_of_is_unit {R A : Type*} [comm_semiring R] [semiring A] [algebra R A]
+  {I : submodule R A} (hI : is_unit I) : I.fg := fg_unit hI.unit
```

Comment:
I'm a little surprised you proved them in this order, but I guess it makes no difference.

Reply:
I'm mostly copying from (refactoring) the original proof for fractional_ideal. But note that both `units.mul_inv` and `units.inv_mul` are used in this proof, while commutativity is used instead in the original proof. This is why proving the `units` version first is more convenient.